keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/resource/ImageHandler.kt

```diff
@@ -49,86 +47,83 @@ class ImageHandler(
   override fun generateName(spec: ImageSpec): ResourceName =
     ResourceName("bakery:image:${spec.artifactName}")
 
-  override fun resolve(resource: Resource<ImageSpec>): ResolvedResource<Image> =
+  override suspend fun desired(resource: Resource<ImageSpec>): Image =
     with(resource) {
       val artifact = DeliveryArtifact(spec.artifactName, DEB)
-      val latestVersion = artifactRepository
-        .versions(artifact)
-        .firstOrNull() ?: throw NoKnownArtifactVersions(artifact)
-      runBlocking {
-        ResolvedResource(
-          desired = desired(resource.spec, latestVersion),
-          current = current(resource.spec, latestVersion)
-        )
-      }
+      val latestVersion = artifact.findLatestVersion()
+      val baseImage = baseImageCache.getBaseImage(spec.baseOs, spec.baseLabel)
+      val baseAmi = findBaseAmi(baseImage)
+      Image(
+        baseAmiVersion = baseAmi,
+        appVersion = latestVersion,
+        regions = spec.regions
+      )
     }
 
-  private suspend fun desired(spec: ImageSpec, version: String): Image {
-    val baseImage = baseImageCache.getBaseImage(spec.baseOs, spec.baseLabel)
-    val baseAmi = findBaseAmi(baseImage)
-    return Image(
-      baseAmiVersion = baseAmi,
-      appVersion = version,
-      regions = spec.regions
-    )
-  }
+  override suspend fun current(resource: Resource<ImageSpec>): Image? =
+    with(resource) {
+      val artifact = DeliveryArtifact(spec.artifactName, DEB)
+      val latestVersion = artifact.findLatestVersion()
```

Comment:
There's some overhead here doing the latest version resolution again but I'd like to revisit this soon anyway. See #313